flink-formats/flink-json/src/test/java/org/apache/flink/formats/json/JsonFormatFactoryTest.java

```diff
@@ -0,0 +1,175 @@
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	private static final TableSchema SCHEMA = TableSchema.builder()
+			.field("field1", DataTypes.BOOLEAN())
+			.field("field2", DataTypes.INT())
+			.build();
+
+	private static final RowType ROW_TYPE = (RowType) SCHEMA.toRowDataType().getLogicalType();
+
+	@Test
+	public void testSeDeSchema() {
+		final Map<String, String> tableOptions = getAllOptions();
+
+		testSchemaSerializationSchema(tableOptions);
+
+		testSchemaDeserializationSchema(tableOptions);
+	}
+
+	@Test
+	public void testFailOnMissingField() {
+		final Map<String, String> tableOptions = getModifyOptions(
+				options -> options.put("json.fail-on-missing-field", "true"));
+
+		thrown.expect(ValidationException.class);
+		thrown.expect(containsCause(new ValidationException("fail-on-missing-field and ignore-parse-errors shouldn't both be true.")));
+		testSchemaDeserializationSchema(tableOptions);
+	}
+
+	@Test
+	public void testInvalidOptionForIgnoreParseErrors() {
+		final Map<String, String> tableOptions = getModifyOptions(
+				options -> options.put("json.ignore-parse-errors", "abc"));
+
+		thrown.expect(ValidationException.class);
+		thrown.expect(containsCause(new IllegalArgumentException("Unrecognized option for boolean: abc. Expected either true or false(case insensitive)")));
+		testSchemaDeserializationSchema(tableOptions);
+	}
+
+	// ------------------------------------------------------------------------
+	//  Utilities
+	// ------------------------------------------------------------------------
+
+	private void testSchemaDeserializationSchema(Map<String, String> options) {
+		final JsonRowDataDeserializationSchema expectedDeser =
+				new JsonRowDataDeserializationSchema(
+						ROW_TYPE,
+						new RowDataTypeInfo(ROW_TYPE),
+						false,
+						true);
+
+		final DynamicTableSource actualSource = createTableSource(options);
+		assert actualSource instanceof TestDynamicTableFactory.DynamicTableSourceMock;
+		TestDynamicTableFactory.DynamicTableSourceMock scanSourceMock =
+				(TestDynamicTableFactory.DynamicTableSourceMock) actualSource;
+
+		DeserializationSchema<RowData> actualDeser = scanSourceMock.sourceValueFormat
+				.createScanFormat(
+						ScanRuntimeProviderContext.INSTANCE,
+						SCHEMA.toRowDataType());
+
+		assertEquals(expectedDeser, actualDeser);
+	}
+
+	private void testSchemaSerializationSchema(Map<String, String> options) {
+		final DynamicTableSink actualSink = createTableSink(options);
+		assert actualSink instanceof TestDynamicTableFactory.DynamicTableSinkMock;
+		TestDynamicTableFactory.DynamicTableSinkMock sinkMock =
+				(TestDynamicTableFactory.DynamicTableSinkMock) actualSink;
+
+		SerializationSchema<RowData> actualSer = sinkMock.sinkValueFormat
+				.createSinkFormat(
+						new SinkRuntimeProviderContext(false),
+						SCHEMA.toRowDataType());
+
+		assertThat(actualSer, instanceOf(JsonRowDataSerializationSchema.class));
```

Comment:
Could you use the same validation as above? I think we will introduce some new options for json seriliazation soon.

Reply:
Woud add `equals` and `hashcode` for the serialization schema.